src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -645,6 +645,8 @@ private static bool TryGetSpecialTypeMethod(SyntaxNode syntax, SpecialMember spe
 
         public override BoundNode VisitTypeOfOperator(BoundTypeOfOperator node)
         {
+            Debug.Assert(node.Type.ExtendedSpecialType == InternalSpecialType.System_Type ||
```

Comment:
>node.Type.ExtendedSpecialType == InternalSpecialType.System_Type [](http://example.com/codeflow?start=25&length=64)

I didn't follow. How do we get to this case (using `InternalSpecialType.System_Type` in typeof)?

Reply:
> I didn't follow. How do we get to this case (using ```InternalSpecialType.System_Type``` in typeof)?

This is a very common case of ```System.Type``` defined in mscorlib. The type of `typeof` is ```System.Type```.

Reply:
Resolved offline, what I was missing was that you can get a type marked as `SpecialType = InternalSpecialType.System_Type` even though we don't have any special-type lookups in the code for it. We use well-known type lookup and generally happen to find the type from corlib, which will be marked as special type.
Thanks for clarification @AlekseyTs